### PR TITLE
missing positions for lambda precondition

### PR DESCRIPTION
**Pull request by jad-hamza**
Some positions were missing in the verification reports ("lambda precondition" checks).

### CODE REVIEW

File: core/src/main/scala/stainless/extraction/PreconditionInference.scala [hunk 1]
```diff
@@ -40,7 +40,10 @@ class PreconditionInference(override val sourceProgram: Program { val trees: ext
       def pre(caller: Expr, args: Seq[Expr]): Expr = caller.getType match {
         case FunctionType(from, to) =>
           val (currArgs, restArgs) = args.splitAt(from.size)
-          and(Application(Pre(caller), currArgs), pre(Application(caller, currArgs), restArgs))
+          and(
+            Application(Pre(caller), currArgs).copiedFrom(caller),
```
Comment (samarion) [inline core/src/main/scala/stainless/extraction/PreconditionInference.scala]: Please add the position to the Pre() tree as well.

File: core/src/main/scala/stainless/extraction/PreconditionInference.scala [hunk 2]
```diff
@@ -40,7 +40,10 @@ class PreconditionInference(override val sourceProgram: Program { val trees: ext
       def pre(caller: Expr, args: Seq[Expr]): Expr = caller.getType match {
         case FunctionType(from, to) =>
           val (currArgs, restArgs) = args.splitAt(from.size)
-          and(Application(Pre(caller), currArgs), pre(Application(caller, currArgs), restArgs))
+          and(
+            Application(Pre(caller), currArgs).copiedFrom(caller), 
+            pre(Application(caller, currArgs).copiedFrom(caller), restArgs)
+          )
         case _ => BooleanLiteral(true)
```
Comment (samarion) [inline core/src/main/scala/stainless/extraction/PreconditionInference.scala]: Set the position as well.

**Comment by jad-hamza** [comment]
I added some more missing positions, but I just noticed that there are still VCs with no positions (e.g. in verification/valid/FlatMap.scala). Investigating!

**Comment by jad-hamza** [comment]
I added some more. And also some in Inox (https://github.com/epfl-lara/inox/pull/38)

I think many are still missing (dozens), but at least now all positions are defined in the verification report for verification/valid/*.scala

**Comment by samarion** [comment]
Merged

**Pull request closed without merging**